src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -142,7 +142,7 @@ end
   f = zeros(MVector{nvariables(equations), eltype(u)})
 
   if orientation == 1
-    f[1] = 0.0
+    f[1] = 0
```

Comment:
Here the assignment of `0` is OK, because `f` is already defined to be a float array and you are just assigning to it 👍